.github/workflows/weekly-release.yml

```diff
@@ -0,0 +1,218 @@
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Collect commits since last release
+        id: changelog
+        run: |
+          LAST_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "")
+          if [ -z "$LAST_TAG" ]; then
+            CHANGELOG_COMMITS=$(git log --pretty="format:- %s" --no-merges HEAD | head -n 10 || true)
+          else
+            CHANGELOG_COMMITS=$(git log --pretty="format:- %s" --no-merges "$LAST_TAG"..HEAD || true)
+          fi
+          if [ -z "$CHANGELOG_COMMITS" ]; then
+            CHANGELOG_COMMITS="- No relevant changes detected since the last release."
+          fi
+          {
+            echo 'commits<<CHANGELOG_EOF'
+            echo "$CHANGELOG_COMMITS"
+            echo 'CHANGELOG_EOF'
+          } >> "$GITHUB_OUTPUT"
+
+      # Generals vc6
+      - name: Download Generals VC6 Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: Generals-vc6+t+e
+          path: generals-vc6-artifacts
+
+      - name: Prepare and Zip Generals VC6
+        run: |
+          mkdir generals-vc6-release
+          cp generals-vc6-artifacts/generalsv.exe generals-vc6-release/GeneralsV.exe
+          cp generals-vc6-artifacts/W3DViewV.exe generals-vc6-release/W3DViewV.exe
+          cp generals-vc6-artifacts/WorldBuilderV.exe generals-vc6-release/WorldBuilderV.exe
+          zip -j generals-preview-${{ needs.get-date.outputs.date }}.zip generals-vc6-release/*
+
+      # GeneralsMD vc6
+      - name: Download GeneralsMD VC6 Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: GeneralsMD-vc6+t+e
+          path: generalsmd-vc6-artifacts
+
+      - name: Prepare and Zip GeneralsMD VC6
+        run: |
+          mkdir generalsmd-vc6-release
+          cp generalsmd-vc6-artifacts/generalszh.exe generalsmd-vc6-release/GeneralsZHv.exe
+          cp generalsmd-vc6-artifacts/W3DViewZH.exe generalsmd-vc6-release/W3DViewZHv.exe
+          cp generalsmd-vc6-artifacts/WorldBuilderZH.exe generalsmd-vc6-release/WorldBuilderZHv.exe
+          zip -j generalszh-preview-${{ needs.get-date.outputs.date }}.zip generalsmd-vc6-release/*
+
+      - name: Generate release notes
+        id: release_body
+        run: |
+          BODY=""
+          if [ "${{ github.event.inputs.build_notes }}" != "" ]; then
+            BODY="${BODY}### Build notes\n${{ github.event.inputs.build_notes }}\n"
+          fi
+          if [ "${{ github.event.inputs.known_issues }}" != "" ]; then
+            BODY="${BODY}### Known issues\n${{ github.event.inputs.known_issues }}\n"
+          fi
+          BODY="${BODY}### Changelog\n${{ steps.changelog.outputs.commits }}"
+          echo "body<<EOF" >> $GITHUB_OUTPUT
+          echo -e "$BODY" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: preview-${{ needs.get-date.outputs.date }}
```

Comment:
Looking at this the format for the release name/tag should be preview-$DATE, but the most recent runs on your repo don't reflect that despite it being 2 weeks since you last pushed the branch? I'm guessing you need to push it to main for that to happen or something? Just wanted to check the output before I gace it the okay.

Also, do we want to call it "preview" which implies its just an early release of what could be final when in practice that won't be the case? Maybe just prefix "weekly" instead? Thoughts?

Reply:
"weekly" is also fine

Maybe "weekly-preview" :D